realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1288,15 +1292,1274 @@ public void add(@Nullable E e) {
         }
     }
 
-    /*
-     * MEMO: This method supports {@code long}, {@code  int}, {@code short}, {@code byte},
-     * {@code double}, {@code float} and {@code boolean} as {@code T} in addition to classes
-     * mentioned in the class comment of {@link RealmValueList}.
-     */
-    @Nonnull
     @Override
-    public <T> T[] toArray(@Nonnull T[] a) {
-        // TODO implement this
+    public <T> T[] toArray(T[] a) {
+        if (!isManaged()) {
+            return super.toArray(a);
+        }
+
+        final Class<?> componentType = a.getClass().getComponentType();
+        final int size = size();
+        if (componentType == String.class) {
+            if (a.length < size) {
+                //noinspection unchecked
+                a = (T[]) Array.newInstance(componentType, size);
```

Comment:
Can we alloc the arrays for all `if` cases? this method is really long.

Reply:
removed special implementation of `toArray(T[])`